src/panels/ExoplanetsPanel/ExoplanetsPanel.tsx

```diff
@@ -48,37 +46,32 @@ export function ExoplanetsPanel() {
     owner!.tags.includes('exoplanet_system')
   );
 
-  const justAdded = loadingAdded.filter(
-    (name) => addedSystems.findIndex((s) => s && s.name.includes(name)) !== -1
-  );
-  const justRemoved = loadingRemoved.filter(
-    (name) => addedSystems.findIndex((s) => s && s.name.includes(name)) === -1
-  );
-  if (justAdded.length > 0) {
-    const newAdded = loadingAdded.filter((e) => !justAdded.includes(e));
-    setLoadingAdded(newAdded);
-  }
-  if (justRemoved.length > 0) {
-    const newRemoved = loadingRemoved.filter((e) => !justRemoved.includes(e));
-    setLoadingRemoved(newRemoved);
+  function isAdded(starName: string) {
+    // Replace all spaces with underscores
+    return (
+      addedSystems.findIndex(
+        (owner) => owner?.identifier === name2Identifier(starName)
+      ) !== -1
+    );
   }
 
-  function removeSystem(starName: string, identifier: string) {
-    // In case we happen to be focused on the removed system star, reset the focus
-    // @TODO (emmbr, 2024-11-29): This will still not check if any of the child nodes are
-    // removed... should be fixed, by setting the anchor/aim property correctly on the
-    // OpenSpace side instead
-    if (anchor === identifier || aim === identifier) {
-      dispatch(setPropertyValue({ uri: NavigationAnchorKey, value: 'Sun' }));
-      dispatch(setPropertyValue({ uri: NavigationAimKey, value: '' }));
-    }
-    luaApi?.exoplanets.removeExoplanetSystem(starName);
-    setLoadingRemoved([...loadingRemoved, starName]);
+  function name2Identifier(starName: string) {
+    return starName.replace(/ /g, '_');
   }
 
-  function addSystem(starName: string) {
-    setLoadingAdded([...loadingAdded, starName]);
-    luaApi?.exoplanets.addExoplanetSystem(starName);
+  function handleClick(starName: string) {
+    const starIdentifier = starName.replace(/ /g, '_');
```

Comment:
The luaApi `makeIdentifier` is too slow as these comparisons happen very often right now. My suggestion is to send a list of identifiers from the engine. Played around with making identifiers in the UI but it just takes way too long

Reply:
Unresolving this to keep the comment visible. :) 

I see, that makes sense. It would likely be solved by a minor redesign of the exoplanets panel though (e.g. moving the "remove" button to only the list of added planets), or keep a list of the redux store. That's outside of the scope of this PR though, so no need to change anything. 

However, this line introduces a potential bug in that the identifier might not match the one for the exoplanet system, so I would at least add a TODO comment mentioning that we should fix it.

Reply:
Also, use the `name2Identifier` funciton here, and add the aforementioned comment over that function instead. To avoid duplicated code
```suggestion
    const starIdentifier = name2Identifier(starName);
```